controller/controller.go

```diff
@@ -339,6 +339,10 @@ func (c *notificationController) processQueueItem() (processNext bool) {
 	appCopy := app.DeepCopy()
 	logEntry := log.WithField("app", key)
 	logEntry.Info("Start processing")
+	subscriptions := c.getSubscriptions(appCopy)
```

Comment:
It is still possible to reproduce the same bug if the application has a subscription but non of the subscriptions triggers a notification. 

The bug is here: https://github.com/argoproj-labs/argocd-notifications/blob/8b9ee894bd976115d30d225d1ba29f378478ef1f/controller/controller.go#L351

If `app.GetAnnotations()` is nil then `appCopy.GetAnnotations()` is empty map. Empty annotations map and nil map should be considered the same.

Reply:
this is no longer required once annotation comparison is fixed